src/main/java/core/tasks/TaskList.java

```diff
@@ -0,0 +1,231 @@
+package core.tasks;
+
+import java.util.ArrayList;
+
+import core.exceptions.EmptyArgumentException;
+import core.exceptions.InvalidDeleteIndexException;
+import core.exceptions.NoTaskToDeleteException;
+import utilities.OutputFormatter;
+
+/**
+ * Class to maintain the list of tasks entered by the user.
+ *
+ * @author s7manth
+ * @version 0.1
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
+
```

Comment:
As suggested in the java coding standard, you could consider using the plural form of the object to represent the collection of objects. ie. `tasks` instead of `taskList` unless there are various collection types for the task object like `taskSet` and `taskList`.